### PR TITLE
Allow mixing lifetime and type parameters

### PR DESCRIPTION
**Pull request by Undin**

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt [hunk 1]
```diff
@@ -168,6 +169,17 @@ private fun checkValueParameter(holder: AnnotationHolder, param: RsValueParamete
     }
 }
 
+private fun checkTypeParameterList(holder: AnnotationHolder, element: RsTypeParameterList) {
+    val lifetimeParams = element.lifetimeParameterList
+    if (lifetimeParams.isEmpty()) return
+    val startOfTypeParams = element.typeParameterList.minBy { it.textOffset }?.textOffset ?: return
```
Comment (matklad) [inline src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt]: Will `.firstOrNull()?.textOffset` work here?

Comment (Undin) [inline src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt]: Can we be sure that list of children in psi element has same order that in code?
If yes then `.firstOrNull()?.textOffset` is the best variant.

Comment (matklad) [inline src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt]: Yep! `getFooList` works by traversing the Ast using `.getFirstChild()`, `.getNextSibling` methods and filtering by `IElementType`, so ordering is preserved. If the ordering was indeterminate, then `Collection` would be used for return type, and not `List`.

Comment (Undin) [inline src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt]: Cool!

Comment (Undin) [inline src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt]: Fixed

**Comment by matklad** [comment]
bors r+

Thanks!

**Pull request merged**